src/Common/src/System/Collections/Generic/SparseArrayBuilder.cs

```diff
@@ -149,8 +149,11 @@ public void CopyTo(T[] array, int arrayIndex, int count)
                 count -= reservedCount;
             }
 
-            // Finish copying after the final marker.
-            _builder.CopyTo(position, array, arrayIndex, count);
+            if (count > 0)
```

Comment:
This is beneficial for scenarios when we insert a marker at the very end, e.g. `nonCollection.Append().Append()./* a bunch of times */` or `nonCollection.Concat(collection)`, so :+1: